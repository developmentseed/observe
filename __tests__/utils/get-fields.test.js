/* global fetch */
/* global test */
/* global __dirname */
/* global expect */

import getFields from '../../app/utils/get-fields'

test('get fields for all field keys', () => {
  const treeFieldKeys = ['leaf_type_singular', 'leaf_cycle_singular', 'denotation']
  const pointFieldKeys = ['name']

  expect(getFields(treeFieldKeys)).toMatchSnapshot()
  expect(getFields(pointFieldKeys)).toMatchSnapshot()
})
