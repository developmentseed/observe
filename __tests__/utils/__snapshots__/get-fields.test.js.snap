// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get fields for all field keys 1`] = `
Array [
  Object {
    "key": "leaf_type",
    "label": "Leaf Type",
    "strings": Object {
      "options": Object {
        "broadleaved": "Broadleaved",
        "leafless": "Leafless",
        "needleleaved": "Needleleaved",
      },
    },
    "type": "combo",
  },
  Object {
    "key": "leaf_cycle",
    "label": "Leaf Cycle",
    "strings": Object {
      "options": Object {
        "deciduous": "Deciduous",
        "evergreen": "Evergreen",
        "semi_deciduous": "Semi-Deciduous",
        "semi_evergreen": "Semi-Evergreen",
      },
    },
    "type": "combo",
  },
  Object {
    "key": "denotation",
    "label": "Denotation",
    "options": Array [
      "agricultural",
      "avenue",
      "urban",
      "part_of_fence",
      "park",
      "street",
      "natural_monument",
      "alley",
      "landmark",
      "cluster",
      "talus",
      "waterway",
      "border_farm",
      "parque",
      "arbres_sur_trottoir",
      "catholic",
      "garden",
      "natural-monument",
      "Avenue",
      "Mauer",
      "shade",
      "alleyhttps://www.klassik-stiftung.de/einrichtungen/schloesser-und-gaerten/schloss-und-park-belvedere/",
      "religious",
      "avenuehttps://www.openstreetmap.org/#map=16/55.8302/-4.2562",
      "roman_catholic",
    ],
    "type": "combo",
  },
]
`;

exports[`get fields for all field keys 2`] = `
Array [
  Object {
    "key": "name",
    "label": "Name",
    "placeholder": "Common name (if any)",
    "type": "localized",
    "universal": true,
  },
]
`;
